pkg/source/markdown.go

```diff
@@ -96,10 +101,11 @@ func (m *markdown) parse(filename string, opts Opts) (err error) {
 	r := bufio.NewReader(f)
 
 	doxIdFound := false
+	doxIgnoreFound := false
 	inComment := false
 	var line string
 	count := 0
-	for count < 2 {
+	for count < 3 {
```

Comment:
Can you help me understand this change?

Reply:
I'd be interested in knowing if either of these cases aren't handled by the original `count < 2` condition.

Reply:
After some testing, I believe you are correct. We can keep the for loop counter set at 2, but we will still need to add `|| count == 2` to the conditional for both parsing the dox ID and the ignore directive. 

With `for count < 2` and without `|| count == 2` I get some errors when parsing markdown files with the cases listed above. 

The following yields `error: EOF`.

```
<!-- dox: 0001 -->
<!-- dox: ignore -->

# Title
```

The following works correctly because it finds the ignore directive first.

```
<!-- dox: ignore -->
<!-- dox: 0001 -->

# Title
```

Having a comment anywhere in the file returns: `error: title not found`

Adding the `|| count == 2` to the conditional makes the above cases I tested pass.